packages/yourloops/components/device/utils/device.utils.ts

```diff
@@ -172,3 +178,11 @@ export const formatParameters = (parameters: ParameterConfig[]): void => {
     parameter.value = formatParameterValue(parameter.value, parameter.unit)
   })
 }
+
+export const flattenParametersChange = (historyParameters: ParametersChange[]): PumpSettingsParameter[] => {
+  let pumpSettingsParameterList: PumpSettingsParameter[] = []
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏This whole logic looks a lot like a `reduce()` 👀